SCFO/Assignment_3/StudentReportSheet/GeneratingOfMarksheet.java

```diff
@@ -0,0 +1,104 @@
+import java.text.DecimalFormat;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+
+public class GeneratingOfMarksheet 
+{
+	ArrayList<StudentData> dataOfStudents= new ArrayList<StudentData>();
+	
+	
+	void userInteractionScreen()
+	{
+		Scanner inr = new Scanner(System.in);
+		System.out.println("Please choose from following options");
+		System.out.println("1.To calculate Maximum, Minimum and Average grades of the students");
+		System.out.println("2.To calculate percentage of passed students");
+		System.out.println("3.Exit");
+		int b = inr.nextInt();
+		if(b>3||b<1)
+		{
+			System.out.println("Your Choice is Invalid");
+			userInteractionScreen();
+		}
+		switch(b)
```

Comment:
This switch should be in else block. Because if you give invalid input, for once, the screen will show the error message and then userInteractionScreen, but when that is over, the control will come back to the switch(b) code try to find something for the invalid input.